### PR TITLE
Expose job properties in environment

### PR DESCRIPTION
**Pull request by frapposelli**
This adds a `$DRONE_BUILD_JOB_NUMBER` variable to the CI environment, set to 0 if the build only has 1 job.

### CODE REVIEW

**Comment by donny-dont** [comment]
LGTM

**Comment by bradrydzewski** [comment]
good call ... I think there are some other JOB fields we can add as well

File: agent/agent.go [hunk 1]
```diff
@@ -255,6 +255,7 @@ func toEnv(w *queue.Work) map[string]string {
 		"DRONE_COMMIT_AUTHOR_EMAIL":  w.Build.Email,
 		"DRONE_COMMIT_AUTHOR_AVATAR": w.Build.Avatar,
 		"DRONE_BUILD_NUMBER":         fmt.Sprintf("%d", w.Build.Number),
+		"DRONE_BUILD_JOB_NUMBER":     fmt.Sprintf("%d", w.Job.Number),
```
Comment (bradrydzewski) [inline agent/agent.go]: let's shorten to `DRONE_JOB_NUMBER` if you don't mind

**Comment by frapposelli** [comment]
@bradrydzewski I've added them to this patch, I realized I needed some of them already :)

**Comment by tboerger** [comment]
LGTM

**Comment by Bugagazavr** [comment]
LGTM

**Comment by bradrydzewski** [comment]
thanks! merging now. I also just updated the docs to reflect this change

**Comment by tboerger** [comment]
@bradrydzewski https://github.com/drone/docs/pull/162

**Comment by bradrydzewski** [comment]
merging now, thanks!

> bradrydzewski drone/docs#162

oops I missed that PR and had already updated the docs, sorry about that

**Pull request merged**